app/src/main/java/com/criteo/pubsdk_android/ServerBiddingActivity.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.pubsdk_android
+
+import android.os.Bundle
+import android.widget.Button
+import android.widget.LinearLayout
+import androidx.appcompat.app.AppCompatActivity
+import com.criteo.publisher.*
+import com.criteo.publisher.model.AdSize
+import com.criteo.publisher.model.BannerAdUnit
+import com.criteo.publisher.model.InterstitialAdUnit
+
+class ServerBiddingActivity : AppCompatActivity() {
+  companion object {
+    val BANNER_ADUNIT = BannerAdUnit(
+        PubSdkDemoApplication.DFP_BANNER_ADUNIT_ID,
+        AdSize(320, 50)
+    )
+
+    val INTERSTITIAL_ADUNIT = InterstitialAdUnit("/140800857/Endeavour_Interstitial_320x480")
+  }
+
+  private lateinit var criteoBannerView: CriteoBannerView
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_server_bidding)
+
+    val buttonBanner: Button = findViewById(R.id.buttonBanner)
+    buttonBanner.setOnClickListener {
+      val bannerViewContainer: LinearLayout = findViewById(R.id.bannerViewContainer)
+      criteoBannerView = CriteoBannerView(this.baseContext, BANNER_ADUNIT)
+      // todo: replace with a fake/demo ajs url
+      criteoBannerView.displayAd("https://ads.eu.criteo.com/delivery/r/ajs.php?did=5f316b617f6e80e2414aa4c068a9a400&u=%7CQnj3nJUtcJBmcy1yUzVZ0sJJVh1R2jYzX03kRoCOGY8%3D%7C&c1=bXzSPZ7XI4X4_eFA_ib-N7k6ayqfn4hlMBTatiYKxuTGVTBcI1VNzEbwEKS44St5EQ4YnwKmKigxwXfKiVfcgEOBtjH0Hh5NsGXtNLl4JkPP8mFeu4Apmc5mfClzCS8yYqxR30zVSw4xDDHw0Vm62BhPaN9RcfzdiS9AbB9nFqbL7um63XLc3suVOCz3emERkg9VaS20rlaR9500TX7BG-L7xYlhaVaZhWmPbvx2xJvLkjNqWiCBpWmejcw7e3my5aWP3cq1MgFY6znEwehZVuF-DflgEeIqGTqhxejhNNl4lZIVFZwtAf1FV1Wo-Kf9gBZejgIknyScKDfX3zha4sfV7hMVU34Btq6evxpZz4Kj9EPSNaPY1EXa3mfcmokLMF6EhZmbHBC_o7tTx5UkBznotTA1Q9ojbajl-oaObczVpHPVdWnjQWl6bPYU2oEoVbwnFR53gWoi_IZU1LME0A")
+      bannerViewContainer.addView(criteoBannerView)
+    }
+
+    val buttonInterstitial: Button = findViewById(R.id.buttonInterstitial)
+    buttonInterstitial.setOnClickListener {
+      val criteoInterstitial = CriteoInterstitial(this.baseContext, INTERSTITIAL_ADUNIT)
```

Comment:
Same comment than above: ad unit is useless, what about adding a constructor without it ?

Reply:
Done. By the way, do you recall the story behind the unused `context`?

Reply:
The context was used before to open the interstitial activity. But now this is handled by the `InterstitialActivityHelper` class which gets its context from the `DependencyProvider`.
Btw, removing the context parameter might be added in the TODO list of breaking changes for next major version.

Reply:
Thanks, clear. Created EE-1249 for that.

Reply:
EE-1229 😉